packages/circuits/src/UnirepCircuit.ts

```diff
@@ -0,0 +1,157 @@
+import path from 'path'
+const snarkjs = require('snarkjs')
+const circom = require('circom')
+import { SnarkProof, SnarkPublicSignals } from '@unirep/crypto'
+
+import { CircuitName } from './types/circuit'
+import { CircuitConfig } from './types/config'
+
+export default class UnirepCircuit {
+    /**
+     * Get circuit config from given zkFilesPath
+     * @param zkFilesPath
+     * @returns circuit config
+     */
+    public static getConfig = (zkFilesPath: string): CircuitConfig => {
+        const circuitConfig: CircuitConfig = require(path.join(
+            zkFilesPath,
+            'config.json'
+        ))
+        return circuitConfig
+    }
+    /**
+     * Get VKey of given circuit
+     * @param zkFilesPath The path of built zk files
+     * @param circuitName The name of circuit
+     * @returns Vkey of the circuit
+     */
+    public static getVKey = (zkFilesPath: string, circuitName: CircuitName) => {
+        const VKeyPath = path.join(zkFilesPath, `${circuitName}.vkey.json`)
+        const vkey = require(VKeyPath)
+        return vkey
+    }
+
+    /**
+     * Get the value of given variable in witness
+     * @param circuit The snarkjs circuit
+     * @param witness The witness of the circuit
+     * @param signal The name of the variable
+     * @returns The value of the variable in witness
+     */
+    public static getSignalByName = (
+        circuit: any,
+        witness: any,
+        signal: string
+    ): any => {
+        return witness[circuit.symbols[signal].varIdx]
+    }
+
+    /**
+     * Compile a .circom file with circom compiler
+     * @param circuitPath The path to the .circom file
+     * @returns The compiled circuit
+     */
+    public static compileAndLoadCircuit = async (circuitPath: string) => {
+        const circuit = await circom.tester(circuitPath)
+        await circuit.loadSymbols()
+
+        return circuit
+    }
+
+    /**
+     * Execute a snarkjs circuit with given inputs
+     * @param circuit The snarkjs circuit
+     * @param inputs Input of the circuit
+     * @returns Witness of the circuit
+     */
+    public static executeCircuit = async (
+        circuit: any,
+        inputs: any
+    ): Promise<any> => {
+        const witness = await circuit.calculateWitness(inputs, true)
+        await circuit.checkConstraints(witness)
+        await circuit.loadSymbols()
+
+        return witness
+    }
+
+    /**
+     * Generate a full proof of the circuit
+     * @param zkFilesPath The path to the built zk files
+     * @param circuitName The name of the circuit
+     * @param inputs The input of the proof
+     * @returns The proof and the public signals of the snark proof
+     */
+    public static genProof = async (
+        zkFilesPath: string,
+        circuitName: CircuitName,
+        inputs: any
+    ) => {
+        const circuitWasmPath = path.join(zkFilesPath, `${circuitName}.wasm`)
```

Comment:
The more i think about this I don't think it's worth doing this in a utility class. Instead we can use the standard snarkjs tooling at the application level.

For example, I changed the unirep social backend to not use the circuits package at all and instead use [this](https://github.com/Unirep/Unirep-Social-backend/blob/anondb-sync/src/daemons/Prover.ts). I think it's good to encourage applications to become more comfortable interacting with zk files.